cmd/oci-image-tool/refs_get.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2016 The Linux Foundation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"encoding/json"
+	"io"
+	"log"
+	"os"
+
+	"github.com/opencontainers/image-spec/image/refs/layout"
+	"github.com/spf13/cobra"
+)
+
+type refsGetCmd struct {
+	stdout io.Writer
+	stderr *log.Logger
+	path   string
+	name   string
+}
+
+func newRefsGetCmd(stdout io.Writer, stderr *log.Logger) *cobra.Command {
+	state := &refsGetCmd{
+		stdout: stdout,
+		stderr: stderr,
+	}
+
+	return &cobra.Command{
+		Use:   "get PATH NAME",
+		Short: "Retrieve a reference from the store",
+		Run:   state.Run,
+	}
+}
+
+func (state *refsGetCmd) Run(cmd *cobra.Command, args []string) {
+	if len(args) != 2 {
+		state.stderr.Print("both PATH and NAME must be provided")
+		if err := cmd.Usage(); err != nil {
+			state.stderr.Println(err)
+		}
+		os.Exit(1)
+	}
+
+	state.path = args[0]
+	state.name = args[1]
+
+	err := state.run()
+	if err != nil {
+		state.stderr.Println(err)
+		os.Exit(1)
+	}
+
+	os.Exit(1)
+}
+
+func (state *refsGetCmd) run() (err error) {
+	engine, err := layout.GetEngine(state.path)
```

Comment:
On Mon, Jun 20, 2016 at 05:32:17AM -0700, Sergiusz Urbaniak wrote:
> > +	engine, err := layout.GetEngine(state.path)
>
> pleas, no getters, see
> https://golang.org/doc/effective_go.html#Getters. Since you
> commented that this method "guesses", just call it `GuessEngine`.

It's not really a getter (in the method to return an instance
attribute sense).  Maybe NewEngine would be more idiomatic for an
instantiator?